packages/moxb/src/util/ShallowStructObservable.ts

```diff
@@ -0,0 +1,92 @@
+import { action, comparer, observable } from 'mobx';
+
+/**
+ * This class provides an observable that tracks attributes at the top level. Similar to the meteor mini mongo:
+ * When a new value is assigned, it changes only the attributes that are structurally different from the current
```

Comment:
Woudn't we want to replace the whole object if the `_id` attribute has changed?

Reply:
I think that happens implicitly -- because we remove attributes that do not exist and add all new attributes, it has the same effect

Reply:
making assumptions about `_id` makes the code less general

Reply:
...and I think if the `_id` changes, then there is often a state in between with an `undefined`

Reply:
This wouldn't really be a limiting assumption, because even if we added special behavior for `_id`, it wouldn't break the main behavior. (The only change would be that it would replace the main object in those cases.)  So it would keep working find with objects without an `_id`.

I think the `_id` use case would be quite common, with objects coming from MongoDB. I would be more certain about this if I knew that when the object is really replaced, everything is fully recalculated.  (There can be side effects that depend on this, etc.)